chain/near.ts

```diff
@@ -0,0 +1,387 @@
+      return changetype<CryptoHash>(this.data as u32)
+    }
+
+    static fromValue(input: string): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.VALUE, changetype<u32>(input))
+    }
+
+    static fromReceiptId(input: CryptoHash): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.RECEIPT_ID, changetype<u32>(input))
+    }
+  }
+
+  export class ExecutionOutcome {
+    constructor(
+      public proof: MerklePath,
+      public blockHash: CryptoHash,
+      public id: CryptoHash,
+      public logs: Array<string>,
+      public receiptIds: Array<CryptoHash>,
+      public gasBurnt: u64,
+      public tokensBurnt: BigInt,
+      public executorId: string,
+      public status: SuccessStatus,
+    ) {}
+  }
+
+  export class SlashedValidator {
+    constructor(
+      public accountId: AccountId,
+      public isDoubleSign: bool,
+    ) {}
+  }
+
+  export class BlockHeader {
+    constructor(
+      public height: BlockHeight,
+      public prevHeight: BlockHeight,// Always zero when version < V3
+      public epochId: CryptoHash,
+      public nextEpochId: CryptoHash,
+      public hash: CryptoHash,
+      public prevHash: CryptoHash,
+      public prevStateRoot: CryptoHash,
+      public chunkReceiptsRoot: CryptoHash,
+      public chunkHeadersRoot: CryptoHash,
+      public chunkTxRoot: CryptoHash,
+      public outcomeRoot: CryptoHash,
+      public chunksIncluded: u64,
+      public challengesRoot: CryptoHash,
+      public timestamp: u64,
+      public timestampNanosec: u64,
+      public randomValue: CryptoHash,
+      public validatorProposals: Array<ValidatorStake>,
+      public chunkMask: Array<bool>,
+      public gasPrice: Balance,
+      public blockOrdinal: NumBlocks,// Always zero when version < V3
+      // public rentPaid: Balance,// TODO: uncomment once in sf proto definition
+      public validatorReward: Balance,
+      public totalSupply: Balance,
+      public challengesResult: Array<SlashedValidator>,
+      public lastFinalBlock: CryptoHash,
+      public lastDsFinalBlock: CryptoHash,
+      public nextBpHash: CryptoHash,
+      public blockMerkleRoot: CryptoHash,
+      public epochSyncDataHash: CryptoHash,// Always empty when version < V3
+      public approvals: Array<Signature>,// Array<Option<Signature>>
+      public signature: Signature,
+      public latestProtocolVersion: ProtocolVersion,
+    ) {}
+  }
+
+  export enum SignatureKind {
+    ED25519 = 0,
+    SECP256K1 = 1,
+  }
+
+  export class Signature {
+    constructor(
+      public kind: SignatureKind,
+      public bytes: Bytes,
+    ) {}
+  }
+
+  export class ValidatorStake {
+    constructor(
+      public accountId: AccountId,
+      public publicKey: PublicKey,
+      public stake: BigInt,
+    ) {}
+  }
+
+  export class ChunkHeader {
+    constructor(
+      public chunkHash: CryptoHash,
+      public signature: Signature,
+      public prevBlockHash: CryptoHash,
+      public prevStateRoot: StateRoot,
```

Comment:
Lets just use `CryptoHash` directly for prevStateRoot.

Reply:
Fixed!